common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -314,10 +314,16 @@ private BaseException getBaseExceptionFromExceptionType(@NonNull final String ex
         );
 
         if (exceptionType.equalsIgnoreCase(UiRequiredException.sName)) {
+            final String errorCode = brokerResult.getErrorCode();
             baseException = new UiRequiredException(
                     brokerResult.getErrorCode(),
                     brokerResult.getErrorMessage()
             );
+            if (OAuth2ErrorCode.INTERACTION_REQUIRED.equalsIgnoreCase(errorCode) ||
+                    OAuth2ErrorCode.INVALID_GRANT.equalsIgnoreCase(errorCode)) {
+                ((UiRequiredException) baseException).setOauthSubErrorCode(brokerResult.getSubErrorCode());
+            }
+
```

Comment:
You may choose to create a helper method called `getUiRequiredException` or `createUiRequiredException` and move this code there and then just invoke that method here.

Reply:
Done!